### PR TITLE
Improve validation tests for unpack builtin functions

### PR DESCRIPTION
**Pull request by haoxli**
The f16 need to enable extra extension, to make the case simpler, I don't want to enable the extension in the return type validation tests. And we have tested f32 and other 32-bits types in the tests, no need f16 here.

Issues: #3618, #3620

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by ben-clayton — COMMENTED** [review]
I think the preferable approach here would be to remove the bespoke `'return'` tests, and instead have `'args'` use `validateConstOrOverrideBuiltinEval` with the the optional return type parameter.
This work is tracked in https://github.com/gpuweb/cts/issues/3613

FYI @amaiorano

**Comment by haoxli** [comment]
Ok, I will update the unpack builtins validation using `validateConstOrOverrideBuiltinEval` with return types.

**Comment by haoxli** [comment]
@ben-clayton  PTAL! Thanks.

File: src/webgpu/shader/validation/expression/call/builtin/unpack2x16float.spec.ts [hunk 1]
```diff
@@ -1,62 +1,126 @@
...
+      [type.create(value)],
+      t.params.stage
+    );
+  });
+
+g.test('arguments')
+  .desc(`Test that ${builtin} is validated correctly when called with different arguments.`)
+  .params(u =>
+    u
+      .combine('arg', keysOf(kArgCases))
```
[the remark this answers is not in the document]
Comment (haoxli) [reply src/webgpu/shader/validation/expression/call/builtin/unpack2x16float.spec.ts:99]: Expect for the `bad_more_args` case, other cases are only one argument for tests.

[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/webgpu/shader/validation/expression/call/builtin/unpack2x16float.spec.ts:99]: Sure, but the value of the `kArgCases` is an array. You have to treat it as if there might be more than one.

File: src/webgpu/shader/validation/expression/call/builtin/unpack2x16float.spec.ts [hunk 2]
```diff
@@ -1,62 +1,126 @@
...
+      .beginSubcases()
+      .expand('returnType', u => (u.arg.includes('good') ? keysOf(kAllValueTypes) : [kReturnType]))
+  )
   .beforeAllSubcases(t => {
-    if (t.params.arg === 'bad_f16') {
+    if (t.params.arg.includes('f16') || t.params.returnType?.toString().includes('f16')) {
```
[the remark this answers is not in the document]
Comment (ben-clayton) [reply src/webgpu/shader/validation/expression/call/builtin/unpack2x16float.spec.ts:104]: Ah, sorry, I don't quite know what I was thinking when I wrote this.

**Review by ben-clayton — APPROVED** [review]

**Pull request merged**